packages/devextreme/js/__internal/ui/chat/typingindicator.ts

```diff
@@ -0,0 +1,190 @@
+import type { Properties as DOMComponentProperties } from '@js/core/dom_component';
+import type { dxElementWrapper } from '@js/core/renderer';
+import $ from '@js/core/renderer';
+import messageLocalization from '@js/localization/message';
+import type { User } from '@js/ui/chat';
+import DOMComponent from '@ts/core/widget/dom_component';
+import type { OptionChanged } from '@ts/core/widget/types';
+
+import Avatar from './avatar';
+
+const CHAT_TYPINGINDICATOR_CLASS = 'dx-chat-typingindicator';
+const CHAT_TYPINGINDICATOR_CONTENT_CLASS = 'dx-chat-typingindicator-content';
+const CHAT_TYPINGINDICATOR_TEXT_CLASS = 'dx-chat-typingindicator-text';
+const CHAT_TYPINGINDICATOR_BUBBLE_CLASS = 'dx-chat-typingindicator-bubble';
+const CHAT_TYPINGINDICATOR_CIRCLE_CLASS = 'dx-chat-typingindicator-circle';
+
+export interface Properties extends DOMComponentProperties {
+  typingUsers: User[];
+}
+
+class TypingIndicator extends DOMComponent<TypingIndicator, Properties> {
```

Comment:
Since the typing bubble shares a similar layout with the MessageGroup component - containing an avatar, text above, and a bubble - it makes sense to reuse the MessageGroup component instead. Did you think about it? What issues did you experience?

Reply:
I'm not going to complicate the already complex logic of the MessageGroup component. These components serve different purposes, they should not be combined.

Reply:
@ksercs Please, resolve the conversation if you agree with Roman

Reply:
I think this is a bad idea because good architecture is based on the principle of sole responsibility. If a MessageGroup component already handles messages, including their display and layout, adding TypingIndicator logic to it can turn it into a ‘Swiss knife’ with different functionality. And as we know, the more functions a Swiss knife has, the less convenient it is to open tin cans)))

Components that perform one function are easier to test, refactor and maintain 🙂

Reply:
I've already set LGTM for the JS implementation - I see it's compact and clear and don't see a significant copypaste taken from MessageGroup. I definitely don't like the idea to use current MessageGroup but try configuring it for this feature purposes: at least because then we'll need to move `_getText` logic to MessageList and it'll break the isolation (or create private options in MessageGroup how Roma described above).

However, I agree that we should try to reuse MessageGroup layout. And I'm even not about code but more about styling. Now I see we have duplicated variables e.g. for bubble color. For me it's more expected that we only add several styles for "circle"/indicator itself, but do nothing for bubble/text etc. Now we'll need to make changes in several places if we decide to update bubble/text appearance. 

So, if you Roma do not have strict reasons why we should keep specific styling for the bubble/text, I'd like we try to refactor it in a separate PR later (branch TypingIndicator from MessageGroup) with DOD - remove all styles except the animation

Reply:
We actually have reasons to add styles specifically for TypingIndicator. Let's discuss this separately in voice.